contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1939,6 +1959,76 @@ check_nullable_identity_constraint(RangeVar *relation, ColumnDef *column)
 						relation->relname)));
 }
 
+/*
+ * Similar to check_invalid_column_constraints, but for individual constraints, such as added by
+ * ALTER TABLE ADD CONSTRAINT
+ */
+static void
+check_invalid_constraints(RangeVar *relation, Constraint *constraint)
+{
+	switch (constraint->contype)
+	{		
+		case CONSTR_DEFAULT:
+		{
+			if (IsA(constraint->raw_expr, FuncCall))
+			{
+				FuncCall *fc = castNode(FuncCall, constraint->raw_expr);
+				if (relation->relpersistence == RELPERSISTENCE_TEMP && !checkAndSetTsqlSystemFunc(fc))
+				{
+					ereport(ERROR,
+							(errcode(ERRCODE_INVALID_COLUMN_DEFINITION),
+							errmsg("User-defined functions, partition functions, and column references are not allowed in expressions in this context.")));
+				}
+			}
+			break;
+		}
+
+		default:
+			break;
+	}
+}
+
+/*
+ * Returns true iff the function is a system function. For Babelfish, this means searching in the "sys"
+ * schema, but the "sys" schema is meant to be opaque to customers, so to match SQL Server behavior
+ * we assume that it will never be schema-qualified in order to be true.
+ * If we do find a matching system function, then we modify the func call to explicitly call the fully-qualified
+ * system function, to prevent inadvertently using any user-defined overrides for the function name.
+ */
+static bool
+checkAndSetTsqlSystemFunc(FuncCall *fc)
+{
+	List *name_to_search;
+	ObjectWithArgs *owa = makeNode(ObjectWithArgs);
+	char *sys = palloc0(4);
+	strncpy(sys, "sys", 3);
+	if (list_length(fc->funcname) == 1)
```

Comment:
this assumes that a function which matches to any sys function name should always resolve to sys function, I am not sure if that is always the case? Does SQL server always give sys functions priority over UDF with same name?

Reply:
yes, because we must specify the schema when invoking a UDF in sql server. anything unqualified means it will search in sys: 

```
1> select foo(1)
2> go
Msg 195, Level 15, State 10, Server 7ecd02cb107a, Line 1
'foo' is not a recognized built-in function name.
1> select dbo.foo(1)
2> go
           
-----------
          1

(1 rows affected)

--- built in func
1> select abs(1)
2> go
           
-----------
          1

(1 rows affected)

```